Configurations/15-ios.conf

```diff
@@ -33,6 +33,12 @@ my %targets = (
     "iossimulator-xcrun" => {
         inherit_from     => [ "ios-common" ],
         CC               => "xcrun -sdk iphonesimulator cc",
+        cflags           => add("-arch x86_64 -mios-simulator-version-min=7.0.0"),
```

Comment:
Note: as expected, this configuration optional currently builds for the host architecture (see recent comments on #21287)

It would be a mistake to change it to always build for `x86_64`.

It is reasonable to add a pair of configuration options `iossimulator-arm64-xcrun` and `iossimulator-x86_64-xcrun`, as long as (per policy) there is a maintainer

Reply:
Those were also my thoughts. My intention was really that the targets should produce the same build on different hosts (arm vs intel).

I don't know the history of these different configs so I didn't dare to change too much. 

However now that you ask would it be okay to change the naming of the targets?

I am proposing something like 
ios-xcrun                              -> ios-arm32-xcrun
ios64-xcrun                         -> ios-arm64-xcrun
iossimulator-xcrun              -> iossimulator-x86_64-xcrun
iossimulator-arm64-xcrun -> iossimulator-arm64-xcrun

And then removing the "cross" configuration targets which to me is quite confusing. 
Adding -isysroot flag can be achieved by `./Configure ios-arm64-xcrun -isysroot $(xcrun --sdk iphoneos --show-sdk-path)` and `./Configure iossimulator-arm64-xcrun -isysroot $(xcrun --sdk iphonesimulator --show-sdk-path)`.

Please let me know what you think @tom-cosgrove-arm.

Reply:
The `iossimulator` targets should build the simulator for the current host, which is what they currently do.

Then to be able to do the cross builds you need to add `iossimulator-x86_64-xcrun` and  `iossimulator-arm64-xcrun`.

If you change any of the existing targets you'll break someone's build, and not just as a side-effect (https://xkcd.com/1172/)

Reply:
I agree that it would be confusing to change behaviour of a config. 

I want to rename the targets to be the following:

ios-arm32-xcrun
ios-arm64-xcrun
iossimulator-xcrun
iossimulator-x86_64-xcrun (new)
iossimulator-arm64-xcrun (new)

Which means I want to delete "iphoneos-cross", "ios-cross" and "ios64-cross" since these can be achieved by appending the configuration call as described above.

Would that be acceptable?

Reply:
> Which means I want to delete "iphoneos-cross", "ios-cross" and "ios64-cross" since
> these can be achieved by appending the configuration call as described above.

As I said:

> If you change any of the existing targets you'll break someone's build

So for now, at least, just add the two iossimulator targets. If you then want to start changing existing targets, I would do those as separate PRs - they are separate issues, and likely to be more contentious, so if you try to do it all in one go you are unlikely to most forward progress

Reply:
That's a good suggestion. I'll update the pull request accordingly.